### PR TITLE
Change default name of attribute/property setters/testers

### PR DESCRIPTION
**Pull request by fingolfin**
Instead of "Tester(FOO)" use "HasFoo", instead of "Setter(FOO)" use "SetFOO".

Err, well, and a few other changes (making up 90% of this PR) that came up when I started to look at that code... :-)

### CODE REVIEW

File: src/opers.c [hunk 1]
```diff
@@ -4895,27 +4895,28 @@ Obj DoVerboseMutableAttribute (
...
-        ImpliedWriteGuard(fname); \
-        ptr = CSTR_STRING(fname); \
-        memcpy( ptr, addon, addon_len ); ptr += addon_len; \
-        *ptr++ = '('; \
-        memcpy( ptr, CSTR_STRING(name), name_len ); ptr += name_len; \
-        *ptr++ = ')'; \
-        *ptr = 0; \
-        RetypeBag( fname, IMMUTABLE_TNUM(TNUM_OBJ(fname)) ); \
-    } while(0)
+Obj WRAP_NAME(Obj name, const char *addon)
```
Comment (stevelinton) [inline src/opers.c]: If the old macro was not visible outside this file, should the new function not be static?

File: src/opers.c [hunk 2]
```diff
@@ -4895,27 +4895,28 @@ Obj DoVerboseMutableAttribute (
...
+{
+    UInt name_len = GET_LEN_STRING(name);
+    UInt addon_len = strlen(addon);
+    Obj fname = NEW_STRING( name_len + addon_len + 2 );
+    ImpliedWriteGuard(fname);
+
+    char *ptr = CSTR_STRING(fname);
+    memcpy( ptr, addon, addon_len ); ptr += addon_len;
```
Comment (stevelinton) [inline src/opers.c]: Split this line (and one similar one below it)?

**Comment by fingolfin** [comment]
@stevelinton I adjusted the code formatting issue you pointed out, and (hopefully) also fixed the failing travis tests (turns out that `KnownAttributesOfObject` and `KnownPropertiesOfObject` hardcoded the assumption that testers are named `Tester(FOO)` -- I fixed that, and changed the comment in the code to hopefully be slightly more helpful than what was there before). Also added one further commit, which cleans up `SetupAttribute`

**Comment by markuspf** [comment]
I can find one place in which `NewAttribute` and `NewProperty` is used, and that already uses `SetX` and `HasX`. This is in `oprt.gd` in calls to `OrbitishOperation"

**Comment by markuspf** [comment]
I think my only mild complaint with the PR as a whole is that, as you say 90% of the changes are tidying. Since we'll be rebasing I don't see this as a problem though; better cleaned up code than not cleaned up code.

**Comment by fingolfin** [comment]
@stevelinton wrote:
> Are there any cases where an attribute is created without using DeclareAttribute, so that the setter and tester do not get bound to the default global names?

You mean code which calls `NewAttribute` (or `NewProperty`) directly? Well, `OrbitsishOperation` does, but it also binds the setters and tester to the expected names.

And then a few packages do this: `sophus`, `fining` and `ToolsForHomalg` -- the latter binds the correct globals, thus is fine.

> This change would actually cause a sort of breakage in that unlikely circumstance.

I am not sure I follow. What do you think might be broken?

**Comment by markuspf** [comment]
The packages `recog`, `fining`, `ToolsForHomalg`, and `sophus` are using `NewProperty` or `NewAttribute` directly.

File: src/opers.c (hunk 1 above)
Comment (fingolfin) [inline src/opers.c]: Yes, I have no added the `static`

**Comment by fingolfin** [comment]
@markuspf just to clarify: do you object to "cleanup for its own sake" (i.e. you prefer me dropping the cleanup), or do you object to this PR mixing it with the actual change (i.e. you'd prefer to have two PRs)? I assume you mean the latter, yes? If so, I can happily split this PR -- then you could merge the cleanup PR first, followed by this PR (which I'd rebase on the other).

However, if the cleanup ends up being accepted anyway, we'll end up with the same result, just in more PRs... *shrug* let me know if you want me to split the PR.

**Comment by fingolfin** [comment]
@markuspf seems we are overlapping here :-). And `recog` does *not* use `NewProperty` and `NewAttribute` -- looking at `grep` output produces a red herring here: they only turn up in unused code (in the "graveyard" `contrib` directory)

**Comment by markuspf** [comment]
@fingolfin I do *not* object to the cleanup, I was just saying that its a bit hidden in this PR, and I am not insisting on splitting it up, since as you say the history will look the same anyway.

**Comment by stevelinton** [comment]
I guess the breakage I was imagining was the function printing as HasX or SetX when it's not actually bound to that global variable. In that case, the output of Print would not be readable to produce the original object, but I admit this is pretty unlikely, especially the library always binds to the Has and Set names.

**Comment by fingolfin** [comment]
@stevelinton ahh, I see! Well, if that happens, I'd consider it a bug in the code using `New*` directly, which should be fixed. IMHO this isn't a blocker for this PR... do you agree?

**Comment by stevelinton** [comment]
Agreed

**Pull request merged**